crates/ra_syntax/src/syntax_node.rs

```diff
@@ -38,14 +38,15 @@ pub type SyntaxElementChildren = rowan::SyntaxElementChildren<RustLanguage>;
 
 pub use rowan::{Direction, NodeOrToken};
 
+#[derive(Default)]
 pub struct SyntaxTreeBuilder {
     errors: Vec<SyntaxError>,
     inner: GreenNodeBuilder<'static>,
 }
 
-impl Default for SyntaxTreeBuilder {
```

Comment:
I am not sure why `Default` trait was implemented here by hand, could you, please, elabortate @matklad?

Reply:
At the time this was written, I forgot to add a `Default` to `GreenBuilder`, and was to lazy to publish a new version of rowan just for that: https://github.com/rust-analyzer/rowan/commit/a3692a9a5ed12f8eff5bb516860b9f2fd2495b6d#diff-1a2b23ceb6e534bd16f63d2e55e537aaR154.

In general, if something looks like it doesn't make sense, it probably indeed doesn't make sense :)

Reply:
I've stumbled with enough amount of things that didn't make sense to me but did actually exist for a reason, so I'd rather ask the author first before amending them. Anyway, thank you for the clarification!